### ISSUE TITLE
[plasma themes apply]: may look different when just applied vs. logging out and in again

### ISSUE THREAD

**Issue by star-buck** [blue-systems/plasma-5.3#51]
themes should fully update to look the way they are (broken or not) right when being applied, and no different after logout and in again

**Comment by eikehein** [comment]
First fix from me: http://commits.kde.org/plasma-desktop/e0d546133656fadc3b536d4b417fe00c13265c04

This fixes the inside panel margin (between panel edge and contents) not updating when switching themes.

In parallel we've analyzed the problem with task buttons (and other things) not receiving visual updates properly when switching themes and Marco has tracked it to a problem in FrameSvgItem's texture generation code; he says he'll address it tomorrow.

We should get this fixed in time for 5.2 overall.

**Comment by notmart** [comment]
the other part is a problem in the textures management of FrameSvgItem, I'll take a look tomorrow

**Comment by notmart** [comment]
The biggest rendering problems should be fixed as well now.
It may still happen that some applets don't immediately update all the elements, but that's because they were using something that doesn't signal, like Svg::hasElement(), so they will be fixed individually as they are found

**Comment by eikehein** [comment]
packages: plasma-desktop, plasma-framework
fixed: Tue 13th
target ISO: Fri 16th
qa method: switch between Breeze and Gnome-Shell-KDE at runtime

**Comment by eikehein** [comment]
Note that the kglobalaccel ticket suggests that the Fri 16th ISO might still have KF5 from the 9th so half the fix might be missing if plasma-framework on the ISO is also from the 9th.

**Comment by star-buck** [comment]
looks good in kci 20150116

**Comment by star-buck** [comment]
still could be improved:
When a theme gets applied, Plasma may look broken due to being "wrongly designed",
then a properly designed theme should at least look the same when applied as if needed to logout and login again to "Fix" the look imo.
Example: install "Midna" Theme from GHNS (which ironically is dubbed "Plasma5" ready) and apply.
theme looks somewhat broken already, after logout and relogin it breaks severly here. Now applying a "well-designed" theme like Oxygen should fix the look without the need to logout and login again, but it stays broken until logged out and in again. Hence the update mechanism doesnt work completely on panel.
Would be great if we could make it work 100% for all properly designed themes and maybe even restore somewhat the look of a Plasma4 theme to not completely break.

**Comment by notmart** [comment]
a problem seems that checking if a particular theme element is present or not can't be done with the qproperties system, so isn't as "magic" as in qml usually is.
This will have to be fixed by hand in every place the problem occurs, to recheck the correct element id when the theme changes.
So far i've identified one problem in the panel and one in the taskbar, will fix today.

**Comment by notmart** [comment]
With last commits of today in plasma-framework and plasma-desktop, the panel and task manager should render correctly right after the theme is changed.
Are there other parts that look broken right after the theme changes?

**Comment by eikehein** [comment]
The TM fix is wrong imho, let's talk on IRC

**Comment by notmart** [comment]
a revised version has been pushed

**Comment by eikehein** [comment]
^ Thanks, TM fix merged into EITM now as well

**Comment by star-buck** [comment]
seems to work better now, closing this then.
